### PR TITLE
Search list

### PR DESCRIPTION
**Pull request by mahjahromi**
[JIRA](https://sensorup.atlassian.net/browse/GEOCENS-21)

## Summary
Search feature is added to "router/views/sensors.vue". It searches through the following fields on each sensor card:
1) MAC (device Id)
2) name 
3) location (city)

## Notes
There is a vue-warning printed in the console "regarding components/side-bar.vue" routing to "profile" which is not defined. 

## Where to start
"filterSensors" function in "router/views/sensors.vue"

### CODE REVIEW

**Review by kanluo — APPROVED** [review]
Looks good to me.

File: src/router/views/sensors.vue [hunk 1]
```diff
@@ -67,6 +82,36 @@ export default {
...
+        (s) => s.location.toLowerCase().indexOf(searchValue) > -1
+      )
+
+      // Merging the results
+      let filteredSensors = filteredMAC
+        .concat(filteredName)
+        .concat(filteredLocation)
+
+      // Removing duplicated values
+      var uniquedFilteredSensors = filteredSensors.concat()
```
Comment (kanluo) [inline src/router/views/sensors.vue]: Could refactor this part by using  `uniqBy` provide by `lodash`( https://lodash.com/docs/4.17.15#uniqBy). It's up to you.

File: src/router/views/sensors.vue [hunk 2]
```diff
@@ -67,6 +82,36 @@ export default {
         return null
       }
     },
+    filterSensors: function() {
+      let searchValue = this.searchInputTxt.toLowerCase().replace(/\s/g, '')
+
+      // Filtering each item (MAC, name and location) based on the input value
+      let filteredMAC = this.sensors.filter(
```
Comment (kanluo) [inline src/router/views/sensors.vue:90]: Could try this Array.prototype.some() 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

**Comment by kanluo** [comment]
It would be nicer if you could put a screenshot in the PR next time. 
And I'll fix the warning in the next PR.

File: src/router/views/sensors.vue (hunk 1 above)
Comment (mahjahromi) [reply src/router/views/sensors.vue]: Sure, I will use uniq

File: src/router/views/sensors.vue (hunk 2 above)
Comment (mahjahromi) [reply src/router/views/sensors.vue:90]: I am not sure if I can use .some for a single value. It is designed for arrays

**Review by mahjahromi — COMMENTED** [review]
I used _.uniq for removing duplicated values of the filter

**Pull request merged**